docs/references/ayx-plugin-cli.md

```diff
@@ -0,0 +1,487 @@
+# AYX Plugin CLI Commands
+
+Explore the [AYX Plugin
+CLI](https://help.alteryx.com/developer-help/ayx-plugin-cli) commands and parameters.
+
+## sdk-workspace-init
+
+This command initializes a workspace---a directory that contains all of
+the source code and configuration details for a set of Alteryx SDK
+plugins. Users create a workspace to house all of their plugin code, and
+once it\'s ready, generate a YXI for said code. The command creates an
+`ayx_workspace.json` file that defines package details and tools. It
+also generates a `backend/` directory for the plugin code, `config/`
+directory for plugin configurations, `ui/` directory for the plugin's UI
+components, a README, and a gitignore file.
+
+Note: Most ayx-plugin-cli commands expect an ayx_workspace.json file in
+the current directory, and throw an exception otherwise.
+
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
+
+#### \--package-name
+
+This parameter becomes the name of the YXI that is generated from the
+workspace.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: Any alphanumeric string.
+
+#### \--tool-category
+
+This parameter is the Designer tool category tab that the resulting
+tools are placed in when the YXI is installed. If the category does not
+exist, one is created when the tool is installed. If left blank, the
+tools are installed under the **Python SDK Examples** category.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+-   Default: Python SDK Examples
+
+#### \--description
+
+The package description.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--author
+
+The plugin author.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--company
+
+The company creating the plugin.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--backend-language
+
+The programming language that a plugin will use. At present, the CLI
+only supports Python plugins.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: python
+
+## create-ayx-plugin
+
+This command generates the boilerplate code, configs, and UI for a new
+Alteryx plugin in the workspace.
+
+Note: If this command fails midway through execution, it leaves
+half-finished files and directories in the workspace. In order to delete
+these, follow these steps:
+
+-   Remove config/\<Plugin\> directory.
+-   Remove backend/ayx_plugins/\<plugin\>.py.
+-   Force-remove ui/\<Plugin\>.
+-   Remove the import statement from \_\_init\_\_.py.
+-   Remove the tool from ayx_workspace.json's "tools" key.
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
```

Comment:
TW review:

Note: If a parameter is not passed in as part of the CLI call, the CLI
waits for standard input for the ignored parameters, even if they're
optional.

Reply:
TODO: Next week in new MR @jcayx